subsys/net/lib/config/init.c

```diff
@@ -379,6 +380,19 @@ int net_config_init_by_iface(struct net_if *iface, const char *app_info,
 #endif
 	}
 
+#if defined(CONFIG_NET_VLAN)
+	if (CONFIG_NET_CONFIG_MY_VLAN_ID > 0) {
```

Comment:
The VLAN ID can be used by the compiler too.

```suggestion
#if defined(CONFIG_NET_VLAN) && CONFIG_NET_CONFIG_MY_VLAN_ID > 0
```

Reply:
yes you are right, is it better with parenthesis?